cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -171,13 +171,24 @@
         (rd WritableReg)
         (rm PReg))
 
-       ;; A MOV[Z,N,K] with a 16-bit immediate.
+       ;; A MOV[Z,N] with a 16-bit immediate.
        (MovWide
         (op MoveWideOp)
         (rd WritableReg)
         (imm MoveWideConst)
         (size OperandSize))
 
+       ;; A MOVK with a 16-bit immediate. Modifies its register; we
+       ;; model this with a seprate input `rn` and output `rd` virtual
+       ;; register, with a regalloc constraint to tie them together.
+       (MovWideMod
+        (op MoveWideOp)
```

Comment:
How about renaming this `Inst` variant to `MovK` (while removing the latter from `MoveWideOp`) and removing this field?

Reply:
Good idea! Done.